migration/mule/service-orchestration-and-choice-routing/README.md

```diff
@@ -0,0 +1,143 @@
+# Service Orchestration and Choice Routing
+
+This example illustrates how the orchestration of Web service calls and asynchronous message processing are utilized to realize an order processing application. With efficient routing and iterative processing, the integration service coordinates with web services hosted by respective vendors to complete the ordering process.
+
+#### Iterative Processing and Aggregating 
+
+Iterative mediator allows us to split a message into a collection of elements and process them iteratively. The iterative mediator has the capability to process the elements parallelly resulting in an efficient mediation flow. Hence, it is recommended when iterative processing involves back-end calls. Aggregator mediator then collects and combines the results of each iteration. In this example, a single order contains multiple items and in order to gather information regarding these individual items, they are processed iteratively. Finally, they are aggregated to calculate the total.
+
+#### Content-Based Routing 
+
+Using the Switch mediator, WSO2 provides the capability to route different messages into different flows of meditation. Switch mediator evaluates a given expression at run time and compares it against the given cases. The expression can be composed using the content of the message such as a certain attribute of the payload or header. If no case matches, the message will be routed to the default case.
```

Comment:
Bold switch mediator. We need to say that we can expression to evaluate a message's properties, or part of its payload  then routes it to a specific sequence (a series of mediators).